cordapp-example/workflows-kotlin/src/main/kotlin/com/example/flow/PaybackFlow.kt

```diff
@@ -0,0 +1,128 @@
+                    status = Vault.StateStatus.UNCONSUMED,
+                    exactParticipants = listOf(ourIdentity, otherParty)
+            ) // */
+
+            //val signedTransaction = serviceHub.validatedTransactions.getTransaction(loanTxId)
+            // val signedTransaction = transactions.find { it.id == transactionHash } ?: throw IllegalArgumentException("Unknown transaction hash.")
+
+            val ourStateRef = StateRef(SecureHash.parse(loanTxId), 0)
+            // val (state, ref) = serviceHub.toStateAndRef<ContractState>(ourStateRef)
+            val inputStateAndRef = serviceHub.toStateAndRef<ContractState>(ourStateRef)
+            val inputIouState = inputStateAndRef.state.data as IOUState;
+
+            // build tx that will consume previous output
+            val txBuilder = TransactionBuilder(notary)
+                    .addInputState(inputStateAndRef)
+                    .addCommand(txCommand)
+
+            // Stage 2.
+            progressTracker.currentStep = VERIFYING_TRANSACTION
+            // Verify that the transaction is valid.
+            txBuilder.verify(serviceHub)
+
+            // Stage 3.
+            progressTracker.currentStep = SIGNING_TRANSACTION
+            // Sign the transaction.
+            val signedTx = serviceHub.signInitialTransaction(txBuilder, inputIouState.lender.owningKey)
+
+            /**
+             * I don't think lender sig is required here,
+             * we should be able to payback without it - but idk how to do `FinalityFlow` without it
+             * also, will it update lender vault and spend IUO if lender will nobe be included in flow?
+             */
+
+            // Send the state to the counterparty, and receive it back with their signature.
+            val otherPartySession = initiateFlow(inputIouState.lender)
+            //val fullySignedTx = subFlow(CollectSignaturesFlow(signedTx, setOf(otherPartySession), ExampleFlow.Initiator.Companion.GATHERING_SIGS.childProgressTracker()))
+
+            // Stage 4.
+            progressTracker.currentStep = FINALISING_TRANSACTION
+            // Notarise and record the transaction in both parties' vaults.
+            return subFlow(FinalityFlow(signedTx, setOf(otherPartySession), FINALISING_TRANSACTION.childProgressTracker()))
+        }
+    }
+
+    @InitiatedBy(Initiator::class)
+    class Acceptor(val otherPartySession: FlowSession) : FlowLogic<SignedTransaction>() {
+        @Suspendable
+        override fun call(): SignedTransaction {
+            val signTransactionFlow = object : SignTransactionFlow(otherPartySession) {
+                override fun checkTransaction(stx: SignedTransaction) = requireThat {
+                    // The transaction involves an IOUState - this ensures that IOUContract will be run to verify the transaction
+                    val output = stx.tx.outputs.single().data
+                    "This must be an IOU transaction." using (output is IOUState)
```

Comment:
To make this a more valid example; try to use `send` in the initiator to send an `Int` and `receive` it here, then validate that input.value = sent value.